[ML] Update documentation regarding versioning. (#98320)
elastic/elasticsearch: docs/reference/transform/apis/upgrade-transforms.asciidoc

[[upgrade-transforms-desc]]
== {api-description-title}

{transforms-cap} are compatible across minor versions and between supported
major versions. However, over time, the format of {transform} configuration
information may change. This API identifies {transforms} which have a legacy
configuration format and upgrades them to the latest version; including clean up
of the internal data structures that store {transform} state and checkpoints.
{transform-cap} upgrade does not affect the source and destination indices.
Note that as of {es} version 8.10.0 a new version number has been introduced to

> track changes to configuration and state in the transform plugin. This

Reviewer: ```suggestion
track changes to configuration and state in the {transform} plugin. This
```

> Note that as of {es} version 8.10.0 a new version number has been introduced to
> track changes to configuration and state in the {transform} plugin. This
> new version number is decoupled from the product version and will increment
> independently as necessary.

Reviewer 2: ```suggestion

NOTE: From {es} 8.10.0,  a new version number is used to
track the configuration and state changes in the {ml} plugin. This new 
version number is decoupled from the product version and will increment
independently.
```
